src/main/java/org/junit/internal/builders/AllDefaultPossibilitiesBuilder.java

```diff
@@ -40,7 +40,10 @@ protected JUnit3Builder junit3Builder() {
     }
 
     protected AnnotatedBuilder annotatedBuilder() {
-        return new AnnotatedBuilder(this);
+        // For compatibility with Request#classes(...) and the Suite runner,
+        // deeper nesting levels always need to support suite() methods:
+        RunnerBuilder suiteBuilder = canUseSuiteMethod ? this : new AllDefaultPossibilitiesBuilder(true);
```

Comment:
If you don't like the additional AllDefaultPossibilitiesBuilder or the additional RunnerBuilder, you can also take this variant, which solves the problem by replacing the bad AllDefaultPossibilitiesBuilder#canUseSuiteMethod with a field that holds the class for which to block the suite() method:
https://github.com/mkeller/junit/commit/5fea7c6e8f8c3cbfc142c0bc62c471e3079a0031